firebase-common/src/main/java/com/google/firebase/StartupTime.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase;
+
+import android.os.SystemClock;
+import androidx.annotation.NonNull;
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+
+/**
+ * Represents the time at which Firebase began initialization, both in unix time/epoch milliseconds
+ * and in milliseconds since boot. The absence of a StartupTime indicates an unreliable or
+ * misleading time, such as a launch in direct boot mode. Because of this, StartupTime cannot be
+ * guaranteed to be present, and instead should be optionally depended on, and its absence handled.
+ */
+@AutoValue
+public abstract class StartupTime {
+
+  public static @Nullable StartupTime getInstance() {
+    return FirebaseApp.getInstance().get(StartupTime.class);
+  }
```

Comment:
I'd avoid coupling this value class to FirebaseApp, as this method is not safe to call when the default app is not initialized. i.e. it will throw IllegalStateException. Ideally SDKs should just inject StartupTime as a component, and given that it's not easy in the case of fireperf, let's not make it very convenient to work around components by providing such static methods.

I suggest introducing such a static method inside the perf sdk, or even just call `FirebaseApp.getInstance().get(StartupTime.class)` directly in places where startup time is needed.

Reply:
Yeah, here's an example of how to silence that error:
https://github.com/firebase/firebase-android-sdk/blob/94ae2ac6fbd7971cbde2ba41824f24248e2ed5e7/firebase-messaging/src/main/java/com/google/firebase/messaging/MessagingAnalytics.java#L198-L199

fyi this is a custom check under https://github.com/firebase/firebase-android-sdk/blob/94ae2ac6fbd7971cbde2ba41824f24248e2ed5e7/tools/errorprone/src/main/java/com/google/firebase/errorprone/ComponentsAppGetCheck.java#L51-L57